File: tests/e2e/runtime/array_overrides/postgresql/java/src/main/java/com/example/db/com/example/db/queries/Querier.java
package com.example.db.queries;

import com.example.db.models.Record;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.Optional;
import javax.sql.DataSource;

public final class Querier {
    private final DataSource dataSource;

    public Querier(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public void insertRecord(String label, java.util.List<java.time.LocalDateTime> timestamps, java.util.List<java.util.UUID> uuids) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            Queries.insertRecord(conn, label, timestamps, uuids);
        }
    }

    public Optional<Record> getRecord(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getRecord(conn, id);
        }
    }
}

File: tests/e2e/fixtures/array_overrides/postgresql/kotlin/src/main/kotlin/com/example/db/com/example/db/queries/Querier.kt
package com.example.db.queries

import com.example.db.models.Record
import javax.sql.DataSource

class Querier(private val dataSource: DataSource) {

    fun insertRecord(label: String, timestamps: List<java.time.LocalDateTime>, uuids: List<java.util.UUID>): Unit =
        dataSource.connection.use { conn -> Queries.insertRecord(conn, label, timestamps, uuids) }

    fun getRecord(id: Long): Record? =
        dataSource.connection.use { conn -> Queries.getRecord(conn, id) }
}
